lib/OptimizationMOI/test/runtests.jl

```diff
@@ -29,6 +31,32 @@ function _test_sparse_derivatives_hs071(backend, optimizer)
     return
 end
 
+@testset "Evaluator" begin
+    rosenbrock(x, p) = (p[1] - x[1])^2 + p[2] * (x[2] - x[1]^2)^2
+    x0 = zeros(2)
+    _p = [1.0, 100.0]
+    cons_circ = (res, x, p) -> res .= [x[1]^2 + x[2]^2]
+    optprob = OptimizationFunction(
+        rosenbrock, Optimization.AutoZygote();
+        cons = cons_circ)
+    prob = OptimizationProblem(optprob, x0, _p, ucons = [Inf], lcons = [0.0])
+    evaluator = init(prob, Ipopt.Optimizer()).evaluator
+
+    x = prob.u0
+    # vector-constraint jacobian product
+    @test (evaluator.f.cons_j !== nothing) || (evaluator.f.cons_jvp !== nothing)
+    y = zeros(1)
+    w = ones(2)
+    @test MathOptInterface.eval_constraint_jacobian_product(evaluator, y, x, w) isa Any
+
+    # constraint jacobian-vector product
+    @test (evaluator.f.cons_j !== nothing) || (evaluator.f.cons_vjp !== nothing)
+    y = zeros(2)
+    w = ones(1)
+    @test MathOptInterface.eval_constraint_jacobian_transpose_product(
+        evaluator, y, x, w) isa Any
```

Comment:
Is it really `Any` here? I guess since it's inplace it returns `nothing`?

Reply:
I guess it should return `nothing`? I just have used `isa Any` on tests that modify inputs to verify that they don't fail. I figured checking the correctness of the return values in `y` was a bit out of scope for this PR.

Reply:
made it explicitly return nothing.